### ISSUE TITLE
txnbuild: add helpers to obtain Claimable Balance IDs

### ISSUE THREAD

**Issue by 2opremio** [stellar/go#3126]
CAP 23 (as part of Stellar Protocol 14 and released with Horizon 1.9.0) , allows creating and claiming Claimable Balances.

Those balances are queried and claimed using their ID which can be obtained:

1. Before submitting the transaction creating the claim. The Claim ID is internally just a Hash which can be inferred even before Core accepts the transaction. A helper method/function would be useful in order to obtain this.
2. After the transaction creating the claim is submitted (i.e. in the transaction result). However, the claimable ID is pretty buried withing the result XDR, making it hard to obtain through the raw API result. A helper would come in handy.

(1) is already addressed by #3122 , (2) still needs to be implemented

**Comment by Shaptic** [comment]
Are we sure that (2) is something we need/want?

- The _creator_ of the claimable balance entry (CB) can use (1). 
- The _recipient_ of the CB must've been either notified 
  - **directly**, e.g. creator sends him the tx, in which case he can also send the balance ID, or
  - **indirectly**, e.g. by polling `/claimable_balances?sponsor=[me]`, in which case the balance ID is directly available in the JSON response

The only scenario that I can think of for (2) is if the recipient is polling `/transactions?from=[sponsor]`, in which case they can literally only see the transaction result XDR, but that just seems.. convoluted.

**Comment by Zagan202** [comment]
I think (2) is a good quality of life feature to have.

To your point @Shaptic, yes it will be kinda redundant if the creator of the CB entry uses (1).
However it is highly plausible that creator doesn't calculate claimable balance ID and send it back directly to the recipient; thus leaving the recipient only the _indirect_ path.
The indirect path ( `polling /claimable_balances?sponsor=[__]`) is tedious when you when the length of the JSON response is greater than 1 entry. 

(2) will ensure that the recipient can obtain the balance ID from the transaction result with ease without relying on the the sender (from any SDK) to forward the balanceID.

**Comment by Shaptic** [comment]
> indirect path is tedious  

This can easily be alleviated with more filters! For example, 
```http
GET /claimable_balances?claimant=[...me...]&order=desc&limit=1
```
would always give you the latest claimable balance entry available to you (if any, of course).

----

The main reason why I brought this up for discussion is that as I was implementing the helper, I realized it was basically the _only_ such helper that we had. That felt strange to me, given that there are plenty of other instances in which it'd be useful to parse things out of the `TransactionResult`, so I had to consider why that's the case.

I do still completely agree that this is a beneficial QoL change; I was just a little quizzical about this being the first time we added a helper like this. Anyway, take a look at #3166 and see if you think it's a good API design!
